### PR TITLE
Make email confirmation check reactive

### PR DESCRIPTION
**Pull request by coreh**
This PR makes the `user` object reactive to email confirmation, and introduces the `EmailVerificationGate` component, to disable portions of the UI if the user is unverified.

### How to Test 

1. Signup for a new account
2. Check that the verification gate is disabling access to the UI on: The menu editor, the invite users page, and on the Settings > Import page.
3. Try resending the email from the unverified notice, make sure that works.
4. Try changing the email from the unverified notice, make sure that sends you to the appropriate user profile page.
5. On a separate window _of the same browser_, open your mailbox and click the link to verify your email address. Make sure that the `EmailVerificationGate` disappears, and the UI is reenabled. (Verifying on a different browser will make the UI take longer to update, since it depends on polling)

Test live: https://calypso.live/?branch=make/email-verification-check-reactive

#### Reactive `user`

We use a combination of polling and signaling across windows (via `localStorage`) to ensure the `user` object gets the `email_verified` property updated as soon as the user confirms their email address.

To avoid unnecessary API requests, polling is only performed by the user object when the user is _not_ verified, at least one React component is subscribed to the `'verify'` event and the window is currently visible. The polling interval is 15 seconds, which is a good compromise between freshness and hitting the API server with too many requests. Most of the time, however, the `localStorage` signaling mechanism will produce an instantaneous update in the user state. (It is triggered by the Email Confirmation success message, whenever `?verified=1` is appended to the URL.)

All components that depend on email confirmation information should subscribe to `'verify'` on mount, and unsubscribe on unmount, to avoid unnecessary polling):

```javascript
class MyComponent extends React.Component {
	// …

	componentWillMount() {
		user.on( 'verify', this.updateVerificationState );
	}

	componentWillUnmount() {
		user.off( 'verify', this.updateVerificationState );
	}
}
```

##### How to use

```jsx
<EmailVerificationGate>
 { /* content here */ }
</EmailVerificationGate>
```

`EmailVerificationGate` wraps the content in a semitransparent div with `pointer-events: none` and logic to disable keyboard focusing of the contents. Since it only blocks the user from accessing the UI, it is complemented with a server side check.

### CODE REVIEW

File: client/components/email-verification/email-unverified-notice.jsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+/**
+ * Internal dependencies
+ */
+
+import userUtils from 'lib/user/utils';
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+import Spinner from 'components/spinner';
+import i18n from 'lib/mixins/i18n';
```
Comment (Tug) [inline client/components/email-verification/email-unverified-notice.jsx]: Hi there, i18n usage has just changed recently. please use:
```
import i18n from 'i18n-calypso';
```
instead

Comment (coreh) [inline client/components/email-verification/email-unverified-notice.jsx]: Fixed.

**Comment by coreh** [comment]
Tested on:

- [x] Safari Technology Preview
- [ ] Safari
- [ ] Chrome
- [ ] IE 11
- [ ] Firefox
- [ ] Edge

**Comment by coreh** [comment]
This PR also switches the email confirmed notice to the Global style, using Redux actions:

**Comment by shaunandrews** [comment]
screen shot 2016-06-10 at 3 19 14 pm

I think the notice on the invite people screen should have some margin on the top/bottom.

Other than that, this is 👍 :shipit:

**Comment by coreh** [comment]
Added the margin back, and tweaked the placement of the `EmailVerificationGate`s, per your suggestion:

screen shot 2016-06-10 at 17 16 23
screen shot 2016-06-10 at 17 16 29
screen shot 2016-06-10 at 17 16 33

**Comment by shaunandrews** [comment]
:+1:

**Comment by jonathansadowski** [comment]
Tested this, and it works great!

It'd be awesome if we could move the email confirmation state to the global state tree... but I realize this isn't a "new data requirement" :)

> All new data requirements should be implemented as part of the global Redux state tree.

from [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md)

I don't know -- what are your thoughts? It seems like we'll be doing enough iterating around this piece that it may be worth moving.

**Comment by coreh** [comment]
A lot of places in the codebase touch `user`, so it would be a kinda big undertaking. Someone with more Redux experience would probably be better at tackling that. (I've only touched a bit of Redux code so far)

Since as you mentioned this doesn't add a new data requirement, but rather just adds polling to an existing property of `user`, I'd rather keep it as it is for now.

We'd also need to figure out a good pattern for doing the polling optimizations with redux (So we only poll while a component is watching).

We can talk to someone more experienced with Redux and decide on a plan for moving `user` over, but since I already have more NUX-focused code code in the pipeline that depends on this, I'd rather merge this one for now.

File: client/components/email-verification/email-verification-gate.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import * as React from 'react';
+
+/**
+ * Internal dependencies
+ */
+
+import EmailUnverifiedNotice from './email-unverified-notice.jsx';
+import userUtils from 'lib/user/utils';
+import _sites from 'lib/sites-list';
+import _user from 'lib/user';
```
Comment (mcsf) [inline client/components/email-verification/email-verification-gate.jsx]: We typically prefer `userFactory` and `sitesFactory` for these two.

**Comment by jonathansadowski** [comment]
> We can talk to someone more experienced with Redux and decide on a plan for moving user over, but since I already have more NUX-focused code code in the pipeline that depends on this, I'd rather merge this one for now.

I would agree. I think if @mcsf's comments are addressed, we could merge this.

File: client/components/email-verification/email-verification-gate.jsx (hunk 2 above)
Comment (coreh) [inline client/components/email-verification/email-verification-gate.jsx]: Fixed.

**Comment by jonathansadowski** [comment]
👍 LGTM

**Pull request merged**